tests/src/gui/testqgsfieldexpressionwidget.cpp

```diff
@@ -225,47 +225,55 @@ void TestQgsFieldExpressionWidget::testFilters()
   std::unique_ptr< QgsFieldExpressionWidget > widget( new QgsFieldExpressionWidget() );
   widget->setLayer( layer );
 
-  QCOMPARE( widget->mCombo->count(), 8 );
-  QCOMPARE( widget->mCombo->itemText( 0 ), QString( "intfld" ) );
-  QCOMPARE( widget->mCombo->itemText( 1 ), QString( "stringfld" ) );
-  QCOMPARE( widget->mCombo->itemText( 2 ), QString( "string2fld" ) );
-  QCOMPARE( widget->mCombo->itemText( 3 ), QString( "longfld" ) );
-  QCOMPARE( widget->mCombo->itemText( 4 ), QString( "doublefld" ) );
-  QCOMPARE( widget->mCombo->itemText( 5 ), QString( "datefld" ) );
-  QCOMPARE( widget->mCombo->itemText( 6 ), QString( "timefld" ) );
-  QCOMPARE( widget->mCombo->itemText( 7 ), QString( "datetimefld" ) );
+  QCOMPARE( widget->mCombo->count(), 9 );
+  QCOMPARE( widget->mCombo->itemText( 0 ), QString( "" ) );
```

Comment:
```suggestion
  QCOMPARE( widget->mCombo->itemText( 0 ), QString() );
```

Reply:
I understand your concerns. QgsFieldExpressionWidget has a default blank value (except if the caller change initial value), but you can't go back to this blank value as soon as you type text. It looks like a strange behavior no? Or maybe I'm missing something. 

Anyway, I can add a *setAllowEmptyFieldName* method in QgsFieldExpressionWidget like it's done in QgsFieldCombobox if your prefer.